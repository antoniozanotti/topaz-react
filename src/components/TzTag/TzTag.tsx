import React from 'react';
import { useVariantStyles } from '../useStyles';

export interface TzTagProps extends React.ComponentProps<"span"> {
  label?: string;
  variant?: 'accent' | 'primary' | 'secondary' | 'negative' | 'dark' | 'light';
  filled?: boolean;
  disabled?: boolean;
}

export function TzTag ({
  label,
  variant = 'accent',
  filled = true,
  disabled = false,
  ...props
}: TzTagProps) {
  /* label Styles */
  let labelOtherStyles = 'rounded font-medium px-2 py-1 text-c2';

  // variants and filled
  let variantStyles = useVariantStyles(variant, filled, false);

  // is disabled
  let labelDisabledStyles = disabled ? 'opacity-50' : '';

  let labelStyles = `${labelOtherStyles} ${variantStyles} ${labelDisabledStyles}`;
  props.className = props.className
    ? labelStyles + " " + props.className
    : labelStyles;

  return (
    <span {...props}>{label}</span>
  );
};
