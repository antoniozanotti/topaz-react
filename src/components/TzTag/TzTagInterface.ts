import { VariantEnum } from "./VariantEnum";

export type TzTagInterface = React.ComponentProps<"span"> & {
  label?: string;
  variant?: keyof typeof VariantEnum;
  filled?: boolean;
  disabled?: boolean;
}