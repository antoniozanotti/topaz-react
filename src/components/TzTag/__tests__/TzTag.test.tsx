import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import { TzTag } from '../TzTag';

const basicTestSpan = (container: HTMLElement) => {
  const span = container.querySelector("span");
  expect(span).toBeInTheDocument();
};

const basicTestLabel = (span: HTMLElement, text: string) => {
  expect(span).toHaveTextContent(text);
};

describe('TzTag Component', () => {
  it('should render span', () => {
    const { container } = render(<TzTag label="Label" />);
    
    basicTestSpan(container);
    basicTestLabel(container, "Label");
    expect(container).toMatchSnapshot();
  });

  it('should render disable state', () => {
    const { container } = render(<TzTag label="Locked" disabled={true} />);
    
    basicTestSpan(container);
    basicTestLabel(container, "Locked");
    expect(container).toMatchSnapshot();
  });
});
