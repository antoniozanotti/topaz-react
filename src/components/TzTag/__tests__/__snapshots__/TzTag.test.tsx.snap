// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TzTag Component should render disable state 1`] = `
<div>
  <span
    class="rounded font-medium px-2 py-1 text-c2  bg-accent dark:bg-dark-accent text-inverse-accent dark:text-inverse-dark-accent opacity-50"
  >
    Locked
  </span>
</div>
`;

exports[`TzTag Component should render span 1`] = `
<div>
  <span
    class="rounded font-medium px-2 py-1 text-c2  bg-accent dark:bg-dark-accent text-inverse-accent dark:text-inverse-dark-accent "
  >
    Label
  </span>
</div>
`;
